qbft/docs/README.md

```diff
@@ -0,0 +1,91 @@
+- `RoundChangeJustification`: used for the Proposal and Round-Change messages. For the Proposal message, it contains a list of Round-Change messages. For the Round-Change message, it contains a list of Prepare messages.
+- `PrepareJustification`: used for the Proposal message. Includes a list of Prepare messages.
+
+Notice that there's no field for the sender (`OperatorID`), the signature or the proposed data (that must be included in the Proposal message and may be included in the Round-Change message). All of this fields are included in the upper wrapper (`types.SignedSSVMessage`). This is useful to detach the whole proposed data from the signature. Notice, however, that the signature is still binded to the proposed data by the field `Root` which is the hash of the data.
+
+## ProcessingMessage
+
+`ProcessingMessage` is an auxiliary structure that holds both a `Message` as well as its upper wrapper `types.SignedSSVMessage`. This is useful because the signature can be verified, using the `types.SignedSSVMessage` field, and the message's content can be easily accessed through the `Message` field without having to do several decoding operations.
+
+## MsgContainer
+
+`MsgContainer` is a structure that holds [`ProcessingMessages`](#processingmessage). It organizes the messages by its round through a map indexed with the `Round` type.
+
+## IConfig
+
+`IConfig` is an interface that allows the configuration of a QBFT execution. For example, it has methods to provide:
+- The leader replica for a certain round (`ProposerF` function)
+- A value check function (similar to the *external predicate* mentioned in the [QBFT paper](https://arxiv.org/pdf/2002.03613)).
+- The time duration for a certain round.
+- The cut-off round (after which the instance should be stopped).
+
+As default implementation, we have the `Config` structure.
+
+## State
+
+`State` is a structure that represents the state of a QBFT instance. It includes:
+- `CommitteeMember`: that contains information about the set of replicas (committee of operators), including their `OperatorIDs` and network public keys.
+- `ID`: an identifier similar to the one hold by the [`Controller`](#controller).
+- `Round`: the current round.
+- `Height`: the height of instance (that identifies the instance). In practice, this `Height` is the slot number of the duty.
+- `LastPreparedRound` and `LastPreparedValue`: the state regarding the last round in which the operator has prepared.
+- `ProposalAcceptedForCurrentRound`: the current round's proposal message, helpful to validate upcoming messages and store the proposed data.
+- `Decided` and `DecidedValue`: stores whether or not the instance has decided and, if so, for which value.
+- Four `MsgContainer` objects, one for each message type.
+
+## Instance
+
+`Instance` is an structure that represents a single QBFT execution. It holds a [`State`](#state), a [`IConfig`](#iconfig), and its start value (to be proposed if possible in case it's the leader).
```

Comment:
> its start value (to be proposed if possible in case it's the leader).

Perhaps worth mentioning what happens to this start value if we are **not a leader**